### PR TITLE
Upgrade Prometheus to master

### PR DESCRIPTION
**Pull request by codesome**
To include https://github.com/prometheus/prometheus/pull/7638

### CODE REVIEW

File: pkg/cortex/modules.go [hunk 1]
```diff
@@ -382,6 +377,9 @@ func (t *Cortex) initQueryFrontend() (serv services.Service, err error) {
 			Reg:        prometheus.DefaultRegisterer,
 			MaxSamples: t.Cfg.Querier.MaxSamples,
 			Timeout:    t.Cfg.Querier.Timeout,
+			NoStepSubqueryIntervalFn: func(int64) int64 {
+				return t.Cfg.Querier.DefaultEvaluationInterval.Milliseconds()
+			},
```
Comment (codesome) [inline pkg/cortex/modules.go:382]: https://github.com/prometheus/prometheus/pull/7628

File: pkg/ruler/ruler.go [hunk 2]
```diff
@@ -509,7 +509,7 @@ func (r *Ruler) syncManager(ctx context.Context, user string, groups store.RuleG
 				level.Error(r.logger).Log("msg", "unable to create rule manager", "user", user, "err", err)
 				return
 			}
-			manager.Run()
+			go manager.Run()
```
Comment (codesome) [inline pkg/ruler/ruler.go:512]: This is now a blocking call

Comment (pracucci) [reply pkg/ruler/ruler.go:512]: Would you mind adding a code comment to explain that the manager `Run()` start running the manager and blocks until the `Stop()` is called, please?

Comment (codesome) [reply pkg/ruler/ruler.go:514]: Done

**Pull request merged**